File: python/videospeedup.py
n, p, k = list(map(int, input().split()))
events = list(map(int, input().split()))

events.append(k)

last_event = 0
orig_length = 0.0
multiplier = 1.0

for i in range(len(events)):
    orig_length += (events[i] - last_event) * multiplier
    multiplier = (100 + (p * (i+1))) / 100
    last_event = events[i]

print(f"{orig_length:.6f}")

File: go/videospeedup/videospeedup.go
package main

import (
	"fmt"
	"io"
)

func main() {
	var n, p, k int
	var next int
	var timestamps []int

	fmt.Scanln(&n, &p, &k)
	
	for {
		n, err := fmt.Scanf("%d", &next)

		if err == io.EOF || n == 0 {
			break
		}

		timestamps = append(timestamps, next)
	}

	timestamps = append(timestamps, k)

	last_event := 0
	orig_length := 0.0
	multiplier := 1.0
	for i, event := range timestamps {
		orig_length += float32(event - last_event) * multiplier
		multiplier = (100 + (float64(p * (i+1)))) / 100
		last_event = event
	}

	fmt.Println(orig_length)
}